cmd/metrics.go

```diff
@@ -202,4 +203,26 @@ func metricsHandler() http.Handler {
 				ErrorHandling: promhttp.ContinueOnError,
 			}),
 	)
+
+}
+
+// AuthMiddleware checks if the bearer token is valid and authorized.
+func AuthMiddleware() func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			config := globalServerConfig.GetPrometheusConfig()
+			if !config.Enable {
+				writeErrorResponse(context.Background(), w, errorCodes.ToAPIErr(ErrPrometheusDisabled), r.URL, guessIsBrowserReq(r))
```

Comment:
We don't need to write XML error response - these are not S3 API operations, we should just write what prometheus client can consume as an error.